File: frontend/src/types/Item.ts
export type ItemDetails = {
  name?: string;
  id?: string;
};

export namespace Item {
  export type Entity = {
    id?: string;
    name?: string;
    note?: string;
    image_url?: string;
    categoryName?: string;
  };

  export type Add = {
    name?: string;
    note?: string;
    categoryName?: string;
    image_url?: string;
  };

  export type LoginResponse = {
    username?: string;
    token?: string;
  };
}

File: Shoppingify/src/main/java/com/whoami/Shoppingify/domain/item/Item.java
package com.whoami.Shoppingify.domain.item;

import com.whoami.Shoppingify.domain.category.Category;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Optional;

@Entity
@Getter
@Setter
@Table(name = "items")
@AllArgsConstructor
@NoArgsConstructor
public class Item {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private  String id;

    @Column(nullable = false)
    private  String name;

    @Column
    private  String note;

    @Column
    private  String image_url;

    @ManyToOne
    @JoinColumn(name = "category_id", nullable = false)
    private  Category category;
}
